[MLIR] Specify new padOp's output type in DropPadUnitDims

dan-garvey opened the pull request:
Previously when dropping unit dim from a pad with mixed dynamic/static input/output shapes, the resulting shape would take on the Type of the input, resulting in invalid IR.

Also did some minor cleanup to the formatting of the `drop_unit_dim_corresponding_to_dynamic_dim` test to make it match the rest of the file.

dan-garvey committed:
[MLIR] Specify outputType in pad op unitDim drop

Previously when dropping unit dim from a pad with mixed
dynamic/static input/output shapes, the resulting shape
would take on the Type of the input, resulting in invalid IR

Additionally, fixed the formatting on the
 drop_unit_dim_corresponding_to_dynamic_dim test in order
to match the rest of the tests in that file.

Signed-off-by: dan <danimal197@gmail.com>

dan-garvey commented:
(The change to the unrelated test was just to make the formatting match the other tests in the file)

banach-space reviewed (approved):
Hey Dan, long time no see!

Thanks for the contribution, LGTM % some nits.

dan-garvey committed:
address comments

dan-garvey commented:
> > If you feel I've adequately addressed your concerns I'd appreciate it if you also hit the merge button for me, as I do not have access yet. = )
> 
> Sure, though I am getting:
> 
> > This commit will be authored by [34486624+dan-garvey@users.noreply.github.com](mailto:34486624+dan-garvey@users.noreply.github.com).
> 
> So, don't forget to follow https://llvm.org/docs/GitHub.html#before-your-first-pr :)

That is a comprehensive doc. I un-privated my email, but I'm not certain if I need to recommit to get it to update. Does it still show the same `authored by`?

banach-space commented:
>  I'm not certain if I need to recommit to get it to update. Does it still show the same authored by?

No, no need for that :)
